flowforge/input/SolidComponents.py

```diff
@@ -0,0 +1,484 @@
+    def height(self) -> float:
+        return self._height
+
+    @height.setter
+    def height(self, height: float):
+        self._height = height
+
+    @property
+    def nCells(self) -> int:
+        return self._n_cells
+
+    @nCells.setter
+    def nCells(self, n_cells: int):
+        self._n_cells = n_cells
+
+    @property
+    def material(self) -> str:
+        return self._material
+
+    @material.setter
+    def material(self, material: str):
+        self._material = material
+
+    @property
+    def azimuthalAngle(self) -> float:
+        assert self.azimuthalAngle == 0.0, "Cannot currently handle non-zero azimuthal angles"
+        return self._azimuthal_angle
+
+    @azimuthalAngle.setter
+    def azimuthalAngle(self, azimuthal_angle: float):
+        self._azimuthal_angle = azimuthal_angle
+
+    @property
+    def zenithAngle(self) -> float:
+        assert self.zenithAngle == 0.0, "Cannot currently handle non-zero zenith angles"
+        return self._zenith_angle
+
+    @zenithAngle.setter
+    def zenithAngle(self, zenith_angle: float):
+        self._zenith_angle = zenith_angle
+
+    @property
+    def crossSection(self) -> CrossSection:
+        return self._cross_section
+
+    @crossSection.setter
+    def crossSection(self, cross_section: CrossSection):
+        assert self._cross_section is None, "Can only add a cross-section object, not alter one."
+        self._cross_section = cross_section
+
+    def _buildCrossSection(self, cross_section_type: str, **kwargs) -> CrossSection:
+        """
+        Given a cross section type and the correct corresponding inputs (via
+        kwargs), this method builds the cross section object used to compute
+        the cross sectional area, as well as the volume
+
+        Parameters
+        ----------
+        cross_section_type : str
+            name of the desired cross section
+
+        Returns
+        -------
+        cross_section : SolidCrossSections.CrossSection
+            Cross section built using the inputs from kwargs
+        """
+        cross_section_obj = cross_section_types[cross_section_type]
+        assert all(inp in kwargs for inp in cross_section_obj.inputs)
+        cross_section = cross_section_obj(**{k: v for k, v in kwargs.items() if k in cross_section_obj.inputs})
+        return cross_section
+
+    def _convertUnits(self, uc: UnitConverter) -> None:
+        """
+        Private method for converting units of the component's internal attribute
+
+        This method is especially useful for converting components to the expected units
+        of the application in which they will be used.
+
+        Parameters
+        ----------
+        uc : UnitConverter
+            A unit converter which holds the 'from' units and 'to' units for the conversion
+            and will ultimately provide the appropriate multipliers for unit conversion.
+        """
+        self.height *= uc.lengthConversion
+        self.crossSection._convertUnits(uc)
```

Comment:
```suggestion
        if not self.crossSection: self.crossSection._convertUnits(uc)
```
Cross section can be None, so check first.

Reply:
Added this check